### PR TITLE
Feature game over state

### PR DESCRIPTION
**Pull request by tomcraven**

### CODE REVIEW

File: src/states/game.js [hunk 1]
```diff
@@ -80,50 +62,110 @@ Game.prototype =
...
+
             this.__non_player_objects.forEach(function(npo_sprite) {
-                npo_sprite.body.position.setTo(npo_sprite.body.position.x -
-                    (ship.body.position.x - ship.body.prev.x),
-                    npo_sprite.body.position.y);
-            });
-            this.offset_x = this.offset_x +
-                (ship.body.position.x - ship.body.prev.x);
-            ship.body.position.setTo(ship.body.prev.x, ship.body.position.y);
+                npo_sprite.body.position.x -= shipXPositionDiff;
```
Comment (marc-jones) [inline src/states/game.js:132]: This line is occasionally throwing an error for me: Cannot read property 'position' of null

Printing __non_player_objects to the console reveals that the array contains a single sprite, then suddenly contains multiple sprites when the error occurs. The number of additional sprites coincides with the number of times you have died and clicked restart, which makes me think that the __non_player_objects array isn't properly deleted between changes of state. Maybe move its initialization back to the 'create' function?

Comment (tomcraven) [inline src/states/game.js:132]: Oh that's a good catch. So it looks like phaser doesn't create a new instance when you switch states..?

Comment (marc-jones) [inline src/states/game.js:132]: I guess not, but what's a bit weird is how it's works fine for a bit and then errors all of a sudden without the player doing anything particularly different. Still, an easy fix even if the cause is still a little hazy!

**Comment by marc-jones** [comment]
All looks good to me!

**Comment by tomcraven** [comment]
schweet, thanks

**Pull request merged**